docs/ApiGuidelines.md

```diff
@@ -14,16 +14,25 @@ Public APIs are the supported APIs that apps can use.
 
 Public API requirements: 
 
-* All public APIs must have at least one test case.
```

Comment:
I would like to add that any new public API should have a design PR that precedes it. Once it is approved, contributors may submit code to implement the new public API.